packages/smart-contracts/scripts-create2/contract-setup/setupERC20SwapToConversion.ts

```diff
@@ -0,0 +1,57 @@
+import { chainlinkConversionPath, erc20SwapConversionArtifact } from '../../src/lib';
+import { HardhatRuntimeEnvironmentExtended } from '../types';
+import utils from '@requestnetwork/utils';
+import { uniswapV2RouterAddresses } from '../../scripts/utils';
+
+/**
+ * Updates the values of the chainlinkConversionPath and swap router of the ERC20SwapToConversion contract
+ * @param contractAddress address of the ERC20SwapToConversion Proxy
+ * @param hre Hardhat runtime environment
+ */
+export const setupERC20SwapToConversion = async (
+  contractAddress: string,
+  hre: HardhatRuntimeEnvironmentExtended,
+): Promise<void> => {
+  // Setup contract parameters
+  const ERC20SwapToConversionContract = new hre.ethers.Contract(
+    contractAddress,
+    erc20SwapConversionArtifact.getContractAbi(),
+  );
+  await Promise.all(
+    hre.config.xdeploy.networks.map(async (network) => {
+      let provider;
+      if (network === 'celo') {
+        provider = new hre.ethers.providers.JsonRpcProvider('https://forno.celo.org');
+        const originalBlockFormatter = provider.formatter._block;
+        provider.formatter._block = (value: any, format: any) => {
+          return originalBlockFormatter(
+            {
+              gasLimit: hre.ethers.constants.Zero,
+              ...value,
+            },
+            format,
+          );
+        };
+      } else {
+        provider = utils.getDefaultProvider(network);
+      }
```

Comment:
Please find a nice way to factorize this, if not the whole if/else, at least the celo part

Reply:
Added the function to the utils package.